demos/readme.md

```diff
@@ -0,0 +1,101 @@
+# Demo apps for the AWS IoT Device SDK for Embedded-C
```

Comment:
Can we capitalize this file name to README.md?

Reply:
Updated.